src/pybind/mgr/dashboard/frontend/src/app/shared/components/copy2clipboard-button/copy2clipboard-button.component.ts

```diff
@@ -41,25 +42,47 @@ export class Copy2ClipboardButtonComponent {
     try {
       const browser = detect();
       const text = this.byId ? this.getText() : this.source;
-      const toastrFn = () => {
-        this.toastr.success('Copied text to the clipboard successfully.');
+      const showSuccess = () => {
+        this.notificationService.show(
+          NotificationType.success,
+          $localize`Success`,
+          $localize`Copied text to the clipboard successfully.`
+        );
+      };
+      const showError = () => {
+        this.notificationService.show(
+          NotificationType.error,
+          $localize`Error`,
+          $localize`Failed to copy text to the clipboard.`
+        );
       };
       if (['firefox', 'ie', 'ios', 'safari'].includes(browser.name)) {
         // Various browsers do not support the `Permissions API`.
         // https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API#Browser_compatibility
-        navigator.clipboard.writeText(text).then(() => toastrFn());
+        navigator.clipboard
+          .writeText(text)
+          .then(() => showSuccess())
+          .catch(() => showError());
       } else {
         // Checking if we have the clipboard-write permission
         navigator.permissions
           .query({ name: 'clipboard-write' as PermissionName })
           .then((result: any) => {
             if (result.state === 'granted' || result.state === 'prompt') {
-              navigator.clipboard.writeText(text).then(() => toastrFn());
+              navigator.clipboard
+                .writeText(text)
+                .then(() => showSuccess())
+                .catch(() => showError());
             }
-          });
+          })
+          .catch(() => showError());
       }
     } catch (_) {
-      this.toastr.error('Failed to copy text to the clipboard.');
+      this.notificationService.show(
+        NotificationType.error,
+        $localize`Error`,
+        $localize`Failed to copy text to the clipboard.`
```

Comment:
Define these as consts and reuse them then, since we have more than one instance of usage for these.